npm_and_yarn/lib/dependabot/npm_and_yarn/package_manager.rb

```diff
@@ -255,22 +252,30 @@ def setup(name)
           )
         end
 
-        version ||= requested_version(name)
-
-        if version
-          raise_if_unsupported!(name, version)
+        if Dependabot::Experiments.enabled?(:enable_corepack_for_npm_and_yarn)
+          version ||= requested_version(name) || guessed_version(name)
 
-          install(name, version)
+          if version
+            raise_if_unsupported!(name, version.to_s)
+            install(name, version)
+          end
         else
```

Comment:
This else is going to be removed in clean-up process of `enable_corepack_for_npm_and_yarn` feature flag after all checks.